WooCommerce/WooCommerceTests/ViewRelated/Orders/Order Creation/ProductSelectorViewModelTests.swift

```diff
@@ -116,25 +116,45 @@ final class ProductSelectorViewModelTests: XCTestCase {
         XCTAssertEqual(viewModel.syncStatus, .empty)
     }
 
+    @MainActor
     func test_sync_status_updates_as_expected_when_products_are_synced() {
         // Given
-        let viewModel = ProductSelectorViewModel(siteID: sampleSiteID, storageManager: storageManager, stores: stores)
-        stores.whenReceivingAction(ofType: ProductAction.self) { action in
+        let mockStorageManager = MockStorageManager()
+        let mockStores = MockStoresManager(sessionManager: .testingInstance)
+        let viewModel = ProductSelectorViewModel(
+            siteID: sampleSiteID,
+            storageManager: mockStorageManager,
+            stores: mockStores
+        )
+        var syncStatusSpy: [ProductSelectorViewModel.SyncStatus] = []
+
+        mockStores.whenReceivingAction(ofType: ProductAction.self) { action in
             switch action {
             case let .synchronizeProducts(_, _, _, _, _, _, _, _, _, _, onCompletion):
-                XCTAssertEqual(viewModel.syncStatus, .firstPageSync)
-                let product = Product.fake().copy(siteID: self.sampleSiteID, purchasable: true)
-                self.insert(product)
+                if let syncStatus = viewModel.syncStatus {
+                    syncStatusSpy.append(syncStatus)
+                }
+                let readOnlyProduct = Product.fake().copy(siteID: self.sampleSiteID, purchasable: true)
+                let product = mockStorageManager.viewStorage.insertNewObject(ofType: StorageProduct.self)
+                product.update(with: readOnlyProduct)
                 onCompletion(.success(true))
             default:
                 XCTFail("Unsupported Action")
             }
         }
 
         // When
-        viewModel.sync(pageNumber: 1, pageSize: 25, onCompletion: { _ in })
+        let _: Bool = waitFor { completion in
+            viewModel.sync(pageNumber: 1, pageSize: 25, onCompletion: { sync in
+                if let syncStatus = viewModel.syncStatus {
+                    syncStatusSpy.append(syncStatus)
+                }
+                completion(sync)
+            })
+        }
```

Comment:
nit: since the boolean value isn't used, it can probably be simplified like
```suggestion
        waitFor { promise in
            viewModel.sync(pageNumber: 1, pageSize: 25, onCompletion: { _ in
                if let syncStatus = viewModel.syncStatus {
                    syncStatusSpy.append(syncStatus)
                }
                promise(())
            })
        }
```

We usually call the callback that we call after the async task is done ["promise"](https://developer.mozilla.org/en-US/docs/Learn/JavaScript/Asynchronous/Promises) in the code base, a concept from web.

Reply:
Gotcha! Changed on https://github.com/woocommerce/woocommerce-ios/pull/9387/commits/349036c3c56effac85b1810a19ea788879af1045